File: app/src/main/java/me/destro/android/gitfav/features/listing/StarredRepositoriesViewModel.java
package me.destro.android.gitfav.features.listing;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import androidx.paging.LivePagedListBuilder;
import androidx.paging.PagedList;

import me.destro.android.libraries.github.model.StarredRepository;
import me.destro.android.gitfav.features.listing.paging.StarredRepositoryFactory;

public class StarredRepositoriesViewModel extends ViewModel {
    LiveData<PagedList<StarredRepository>> mStarredRepository;

    public LiveData<PagedList<StarredRepository>> getStarredRepositories(String username) {
        this.mStarredRepository = new LivePagedListBuilder<>(new StarredRepositoryFactory(username), 20)
                .build();
        return this.mStarredRepository;
    }
}

File: app/src/main/java/me/destro/android/gitfav/features/listing/StarredRepositoriesViewModel.kt
package me.destro.android.gitfav.features.listing

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.paging.LivePagedListBuilder
import androidx.paging.PagedList
import me.destro.android.gitfav.data.repository.RemoteRepository
import me.destro.android.gitfav.domain.model.Repository

import me.destro.android.gitfav.features.listing.paging.StarredRepositoryFactory

class StarredRepositoriesViewModel(private val remoteRepository: RemoteRepository) : ViewModel() {
    private lateinit var mStarredRepository: LiveData<PagedList<Repository>>

    fun getStarredRepositories(username: String): LiveData<PagedList<Repository>> {
        this.mStarredRepository = LivePagedListBuilder(StarredRepositoryFactory(username, remoteRepository), 20)
                .build()
        return this.mStarredRepository
    }
}
